src/gsy_e/models/area/__init__.py

```diff
@@ -152,12 +149,124 @@ def name(self, new_name):
+        if self.parent is not None:
+            grid_fee_constant = self.grid_fee_constant if self.grid_fee_constant else 0
+            return grid_fee_constant + self.parent.get_path_to_root_fees()
+        return self.grid_fee_constant if self.grid_fee_constant else 0
+
+    def get_grid_fee(self):
+        """Return the current grid fee for the area."""
+        grid_fee_type = (
+            self.config.grid_fee_type if self.config is not None
+            else ConstSettings.MASettings.GRID_FEE_TYPE)
+
+        return self.grid_fee_constant if grid_fee_type == 1 else self.grid_fee_percentage
+
+    def update_config(self, **kwargs):
+        """Update the configuration of the area using the provided arguments."""
+        if not self.config:
+            return
+        self.config.update_config_parameters(**kwargs)
+        if self.strategy:
+            self.strategy.read_config_event()
+        for child in self.children:
+            child.update_config(**kwargs)
+
     def get_state(self):
         """Get the current state of the area."""
         state = {}
         if self.strategy is not None:
             state = self.strategy.get_state()
 
+        return state
+
+    def restore_state(self, saved_state):
+        """Restore a previously-saved state."""
+        if self.strategy is not None:
+            self.strategy.restore_state(saved_state)
+
+    def area_reconfigure_event(self, **kwargs):
+        """Reconfigure the device properties at runtime using the provided arguments."""
+        if self.strategy is not None:
+            self.strategy.area_reconfigure_event(**kwargs)
+            return True
+
+        grid_fee_constant = (
+            kwargs["grid_fee_constant"]
+            if key_in_dict_and_not_none(kwargs, "grid_fee_constant")
+            else self.grid_fee_constant)
+        grid_fee_percentage = (
+            kwargs["grid_fee_percentage"]
+            if key_in_dict_and_not_none(kwargs, "grid_fee_percentage")
+            else self.grid_fee_percentage)
+
+        try:
+            validate_area(grid_fee_constant=grid_fee_constant,
+                          grid_fee_percentage=grid_fee_percentage)
+
+        except (GSyAreaException, GSyDeviceException) as ex:
+            log.error(ex)
+            return None
+
+        self.update_descendants_strategy_prices()
+        return None
+
+    def update_descendants_strategy_prices(self):
+        """Recursively update the strategy prices of all descendants of the area."""
+        try:
+            if self.strategy is not None:
+                self.strategy.event_activate_price()
+            for child in self.children:
+                child.update_descendants_strategy_prices()
+        except GSyException:
+            log.exception("area.update_descendants_strategy_prices failed.")
+            return
+
+
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=too-many-public-methods
+class Area(AreaBase):
```

Comment:
In order to limit the disabling only to the class scope, please move these inside of the class.
Or did I get this wrong before?

Reply:
Will fix this, it was like this before IIRC, that's why it went under my radar. I have the same recollection too, that in order for the pylint disables to be applied in the class scope they have to be declared inside the class.